src/IIIFingest/tests/functional_tests/test_bucket.py

```diff
@@ -0,0 +1,55 @@
+import os.path
+
+import boto3
+import pytest
+from boto3.exceptions import S3UploadFailedError
+from botocore.exceptions import ClientError
+from dotenv import load_dotenv
+
+from ...bucket import upload_directory, upload_image_get_metadata
+from ...settings import ROOT_DIR
+
+load_dotenv()
+
+
+class TestFunctionalBucket:
+    abs_path = os.path.abspath(os.path.join(ROOT_DIR, '../..'))
+    image_path = os.path.join(abs_path, "images", "27.586.1-cm-2016-02-09.tif")
+    s3_path = "testing/"
+    file_name = "27.586.1-cm-2016-02-09.tif"
+    bucket_name = os.getenv('TEST_BUCKET')
+    key = f"{s3_path}{file_name}"
+    default_session = boto3._get_default_session()
+    image_dir_path = os.path.join(abs_path, "images")
```

Comment:
For functional tests, maybe it makes sense to allow the user to also set the AWS_PROFILE / keys in `.env`, and then create and use a new session with those? First time I tried running this I wasn't in the right profile and it failed, took me a minute to figure out why.

Reply:
Definitely agreed with allowing the user to set up the default AWS_PROFILE to their desired profile for functional testing. Will update the `.env` with those changes!

Reply:
Note: I only added the `TEST_ACCESS_KEY` and `TEST_SECRET_KEY` to the `.env` since I didn't think `AWS_PROFILE` was needed for these test.

Reply:
Sounds good - I think the other approach would be to use the named profiles in the shared credentials file at `~/.aws/credentials`, instead of duplicating those keys to `.env`. This also reduces the impact if `.env` was committed despite the `.gitignore`. [Docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#shared-credentials-file)

```
import boto3

session = boto3.Session(profile_name='mps-atmch-dev')
dev_s3_client = session.client('s3')
```

Reply:
Sounds good! Will make those changes!